lisa/tools/ntttcp.py

```diff
@@ -444,6 +469,151 @@ def create_ntttcp_udp_performance_message(
             other_fields,
         )
 
+    def _send_unified_perf_metrics(
+        self,
+        metrics: List[Dict[str, Any]],
+        test_case_name: str,
+        test_result: "TestResult",
+    ) -> None:
+        """Helper method to send unified performance metrics."""
+        tool = constants.NETWORK_PERFORMANCE_TOOL_NTTTCP
+        
+        for metric in metrics:
+            send_unified_perf_message(
+                node=self.node,
+                test_result=test_result,
+                test_case_name=test_case_name,
+                tool=tool,
+                metric_name=metric["name"],
+                metric_value=metric["value"],
+                metric_relativity=metric["relativity"],
+            )
+
+    def send_ntttcp_tcp_unified_perf_messages(
+        self,
+        server_result: NtttcpResult,
+        client_result: NtttcpResult,
+        latency: Decimal,
+        connections_num: str,
+        buffer_size: int,
+        test_case_name: str,
+        test_result: "TestResult",
+    ) -> None:
+        """Send unified performance messages for TCP ntttcp metrics."""
+        metrics = [
+            {
+                "name": "throughput_in_gbps",
+                "value": float(client_result.throughput_in_gbps),
+                "relativity": MetricRelativity.HigherIsBetter,
+            },
+            {
+                "name": "latency_us",
+                "value": float(latency),
+                "relativity": MetricRelativity.LowerIsBetter,
+            },
+            {
+                "name": "connections_num",
+                "value": int(connections_num),
+                "relativity": MetricRelativity.NA,
+            },
+            {
+                "name": "buffer_size",
+                "value": float(buffer_size),
+                "relativity": MetricRelativity.NA,
+            },
+            {
+                "name": "retrans_segments",
+                "value": float(client_result.retrans_segs),
+                "relativity": MetricRelativity.LowerIsBetter,
+            },
+            {
+                "name": "connections_created_time",
+                "value": float(client_result.connections_created_time),
+                "relativity": MetricRelativity.LowerIsBetter,
+            },
+            {
+                "name": "rx_packets",
+                "value": float(server_result.rx_packets),
+                "relativity": MetricRelativity.HigherIsBetter,
+            },
```

Comment:
Set `protocol_type` of the base class `PerfMessage` to tcp or udp, so the protocol type is saved for messages.

Reply:
@LiliDeng protocol_type  is in the PergMessageBase already. let's just keep it as is.